### PR TITLE
Refactor loading genomics

### PR DESCRIPTION
**Pull request by CunliangGeng**
This PR intends to refactor the loading of strain mappings for genomics part, which involves the loading of not only the strain mappings but also genomics data (e.g. BGC, GCF).

The major changes:
- Update the logics of  `DatasetLoader._load_genomics` method with newly created/refactored functions/classes
- Update the logics of `StrainCollection` methods `add`, `remove` and `lookup`
- Update or add methods to classes `GCF`, `MibigBGCLoader`, `Strain` and `StrainCollection`

Expected Failed test: `test_load_metabolomics`

### CODE REVIEW

File: tests/test_strain.py [hunk 1]
```diff
@@ -26,10 +26,16 @@ def test_hash(strain: Strain):
     assert hash(strain) == hash("strain_1")
 
 
+def test_names(strain: Strain):
+    assert strain.names == {"strain_1", "strain_1_a"}
```
Comment (gcroci2) [inline tests/test_strain.py:30]: Where is the alias added? I see that in the Strain class it defaults to an empty set. 
Also for test_repr and test_str, you assert that there is 1 alias, but the length of _aliases should be 0, or am I wrong?

File: src/nplinker/strain_collection.py [hunk 2]
```diff
@@ -55,28 +55,46 @@ def add(self, strain: Strain) -> None:
...
+                if name not in self._strain_dict_name:
+                    self._strain_dict_name[name] = [strain]
+                else:
+                    self._strain_dict_name[name].append(strain)
 
     def remove(self, strain: Strain):
         """Remove a strain from the collection.
 
+        It removes the given strain object from the collection by strain id.
+        If the strain id is not found, it does nothing.
```
Comment (gcroci2) [inline src/nplinker/strain_collection.py]: Should we log a warning in this case?

File: tests/test_strain_collection.py [hunk 3]
```diff
@@ -43,14 +43,84 @@ def test_iter(collection: StrainCollection, strain: Strain):
...
     sut.add(strain)
+    assert len(sut) == 1
     assert strain in sut
     for name in strain.names:
         assert name in sut._strain_dict_name
 
 
-def test_remove(collection: StrainCollection, strain: Strain):
+def test_add_same_id_same_alias(strain: Strain, collection: StrainCollection):
+    collection.add(strain)
```
Comment (gcroci2) [inline tests/test_strain_collection.py:53]: General question about pytest: when you pass in a var like `strain`, type hinting that it's a `Strain` object (as above), does pytest instantiate it internally with the defaults? `collection` is defined above as a fixture, so it's clear, but `strain` it's not so I was wondering this

**Review by gcroci2 — APPROVED** [review]
I left very minor comments :)

File: tests/test_strain.py (hunk 1 above)
Comment (CunliangGeng) [reply tests/test_strain.py:30]: The `strain` is a shared fixture defined in the `tests/conftest.py`. There is one alias for the strain with name `strain_1_a`.

File: tests/test_strain_collection.py (hunk 3 above)
Comment (CunliangGeng) [reply tests/test_strain_collection.py:53]: The strain is a shared fixture defined in the `tests/conftest.py`.

File: src/nplinker/strain_collection.py (hunk 2 above)
Comment (CunliangGeng) [reply src/nplinker/strain_collection.py]: good catch, we should raise a ValueError actually. Added in new commit.

File: tests/test_strain.py (hunk 1 above)
Comment (gcroci2) [reply tests/test_strain.py:30]: Oh now I saw it and everything makes sense :-)

**Review by gcroci2 — APPROVED** [review]

**Pull request merged**